core/whistle_number_manager-1.0.0/src/wh_port_request_crawler.erl

```diff
@@ -0,0 +1,152 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Peter Defebvre
+%%%-------------------------------------------------------------------
+-module(wh_port_request_crawler).
+
+-behaviour(gen_server).
+
+%% API
+-export([start_link/0
+         ,stop/0
+        ]).
+
+%% gen_server callbacks
+-export([init/1
+         ,handle_call/3
+         ,handle_cast/2
+         ,handle_info/2
+         ,terminate/2
+         ,code_change/3
+        ]).
+
+-include("wnm.hrl").
+
+-define(MOD_CONFIG_CAT, <<(?WNM_CONFIG_CAT)/binary, ".port_request">>).
+
+-record(state, {}).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Starts the server
+%%
+%% @spec start_link() -> {ok, Pid} | ignore | {error, Error}
+%% @end
+%%--------------------------------------------------------------------
+start_link() ->
+    gen_server:start_link(?MODULE, [], []).
+
+stop() ->
+    self() ! 'stop'.
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Initializes the server
+%%
+%% @spec init(Args) -> {ok, State} |
+%%                     {ok, State, Timeout} |
+%%                     ignore |
+%%                     {stop, Reason}
+%% @end
+%%--------------------------------------------------------------------
+init([]) ->
+    put('callid', ?MODULE),
+    lager:debug("started ~s", [?MODULE]),
+    self() ! 'start_crawling',
```

Comment:
For reference, you can view konami_event_listener's use of `cleanup_timer/0` in `init/1` and `handle_info/2`